### PR TITLE
Refactor reports and new feature

### PR DESCRIPTION
**Pull request by PauliusLu**
Changed the code in `AdminController.cs` to be more readable.
Implemented full CRUD (Create, read, update and delete) support for managing Reports.
Added that if Report is accepted, all Reports for the same Post is also accepted.

P.S. Don't forget to do update-database

### CODE REVIEW

File: Karma/Controllers/AdminController.cs [hunk 1]
```diff
@@ -338,11 +339,122 @@ public async Task<IActionResult> ReportReview(int reportId, string confirm, stri
...
 
+        [HttpDelete]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> ReportReview(int reportId)
+        {
+
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToPage("/Account/Login", new { area = "Identity" });
+            }
```
Comment (ArmintasP) [inline Karma/Controllers/AdminController.cs:350]: ref1.

File: Karma/Controllers/AdminController.cs [hunk 2]
```diff
@@ -338,11 +339,122 @@ public async Task<IActionResult> ReportReview(int reportId, string confirm, stri
...
+                page = page,
+                sorting = sorting,
+                aggr = aggr
+            };
+
+
+            if (aggr == "on")
+                reportsDM.Reports = GetAggregatedReports(reports, posts, reportsPerPage, rptState, filter, page, viewData);
+            else
+                reportsDM.Reports = GetPagedReports(reports, posts, reportsPerPage, rptState, filter, page, viewData);
```
Comment (ArmintasP) [inline Karma/Controllers/AdminController.cs:406]: See ref1. You've used curly braces even for one line, but here you are avoiding it here. Some consistency would be nice. In addition, it is easier to edit the code and avoid mistakes in those blocks if somebody writes additional lines after if/else.

**Review by ArmintasP — COMMENTED** [review]
LGTM.

File: Karma/Controllers/AdminController.cs (hunk 2 above)
Comment (PauliusLu) [reply Karma/Controllers/AdminController.cs:406]: The reason why I removed curly braces was that the function didn't have much code, but was quite long. And yeah, I agree, consistency would be nice. So I'll add curly braces.

**Review by TomB-0 — APPROVED** [review]
LGTM

**Review by baikauskaite — APPROVED** [review]

**Pull request merged**